### DISCUSSION TITLE
Telling me to run commands as "next steps" - where and how?

### DISCUSSION THREAD

**Discussion by jenniearll** [Homebrew/discussions#4741] [Getting started]
### Output of `brew config`

```shell
I have no idea what you asking
```

### Output of `brew doctor`

```shell
Still have no idea
```

### Description of issue

I am just learning coding and was trying to install homebrew. This is the first time I have installed anything on the terminal so I am not familiar with working with it. I received the following message and I have no idea what to do with this:

==> Next steps:
- Run these two commands in your terminal to add Homebrew to your PATH:
    (echo; echo 'eval "$(/usr/local/bin/brew shellenv)"') >> /Users/TJ/.profile
    eval "$(/usr/local/bin/brew shellenv)"
- Run brew help to get started

Can anyone help me figure out what I need to do here? Where do I run these commands? Which parts are the actual commands? How? How do I run brew help? Thank you!

**Comment by hedbonker** [comment]
In terminal window copy this from instructions and paste into terminal window:
(echo; echo 'eval "$(/usr/local/bin/brew shellenv)"') >> /Users/TJ/.profile eval "$(/usr/local/bin/brew shellenv)"

Do the same for the next command:

brew help

or instead of copy/paste, just type them into the terminal window - hit enter at the end of each command.

**Reply by Selade1** [reply]
i feel so dumb lmao thanks kind sir

**Comment by jenniearll** [comment]
Thank you! I did that and received the following:

-bash: syntax error near unexpected token `eval'


So, did it run correctly or is there an error?

On Fri, Aug 25, 2023 at 12:56 PM hedbonker ***@***.***> wrote:

> In terminal window copy this from instructions and paste into terminal
> window:
> (echo; echo 'eval "$(/usr/local/bin/brew shellenv)"') >>
> /Users/TJ/.profile eval "$(/usr/local/bin/brew shellenv)"
>
> Do the same for the next command:
>
> brew help
>
> or instead of copy/paste, just type them into the terminal window - hit
> enter at the end of each command.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/Homebrew/discussions/4741#discussioncomment-6825531>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BCESGMGJPPNO7RYV6CKQ63DXXDRLHANCNFSM6AAAAAA36Y4RVQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Jenni Earll
***@***.***
http://www.linkedin.com/in/jenniearll

**Reply by osalbahr** [reply]
Could you copy paste the command you ran? You might’ve mistyped a character. I recommend copying the command displayed to you as-is instead of typing it.

**Comment by jenniearll** [comment]
Hi! I copied and pasted exactly what hedbonker suggested, I did not type it
out. So this is what I ran:

(echo; echo 'eval "$(/usr/local/bin/brew shellenv)"') >> /Users/TJ/.profile
eval "$(/usr/local/bin/brew shellenv)"


I am considering uninstalling and reinstalling Homebrew but I do not see
how that will fix the problem, honestly.

On Fri, Aug 25, 2023 at 1:17 PM Osama Albahrani ***@***.***>
wrote:

> Could you copy paste the command you ran? You might’ve mistyped a
> character. I recommend copying the command displayed to you as-is instead
> of typing it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/Homebrew/discussions/4741#discussioncomment-6825699>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BCESGMHELRRIEGYBYD6OVHDXXDT2NANCNFSM6AAAAAA36Y4RVQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Jenni Earll
***@***.***
http://www.linkedin.com/in/jenniearll

**Reply by osalbahr** [reply]
Could you try running the command again and copy-paste the command *and* the error in one go?

**Comment by jenniearll** [comment]
Sure, here it is:

(echo; echo 'eval "$(/usr/local/bin/brew shellenv)"') >> /Users/TJ/.profile
eval "$(/usr/local/bin/brew shellenv)"

-bash: syntax error near unexpected token `eval'

On Fri, Aug 25, 2023 at 4:33 PM Osama Albahrani ***@***.***>
wrote:

> Could you try running the command again and copy-paste the command *and*
> the error in one go?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/Homebrew/discussions/4741#discussioncomment-6827078>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BCESGMEHQB7L7VGV6DOQJSDXXEK4JANCNFSM6AAAAAA36Y4RVQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Jenni Earll
***@***.***
http://www.linkedin.com/in/jenniearll

**Reply by osalbahr** [reply]
That’s odd. Does the error also happen if you run the first command (first line) alone, hit enter, then enter the second command?

**Comment by jenniearll** [comment]
And for reference, here is exactly the message from the initial install:

*==>** Next steps:*

- Run these two commands in your terminal to add Homebrew to your *PATH*:

    (echo; echo 'eval "$(/usr/local/bin/brew shellenv)"') >>
/Users/TJ/.profile

    eval "$(/usr/local/bin/brew shellenv)"

On Fri, Aug 25, 2023 at 4:58 PM Jenni Earll ***@***.***> wrote:

> Sure, here it is:
>
> (echo; echo 'eval "$(/usr/local/bin/brew shellenv)"') >>
> /Users/TJ/.profile eval "$(/usr/local/bin/brew shellenv)"
>
> -bash: syntax error near unexpected token `eval'
>
> On Fri, Aug 25, 2023 at 4:33 PM Osama Albahrani ***@***.***>
> wrote:
>
>> Could you try running the command again and copy-paste the command *and*
>> the error in one go?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/orgs/Homebrew/discussions/4741#discussioncomment-6827078>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/BCESGMEHQB7L7VGV6DOQJSDXXEK4JANCNFSM6AAAAAA36Y4RVQ>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>
>
> --
> Jenni Earll
> ***@***.***
> http://www.linkedin.com/in/jenniearll
>


-- 
Jenni Earll
***@***.***
http://www.linkedin.com/in/jenniearll

**Comment by jenniearll** [comment]
First, thank you so much for your help. Can you walk me through this?
Exactly which part is the first command and which is the second? I can't
tell where one ends and the other begins and if everything there is part of
the commands or not. This is the first time I've ever looked at the
terminal of my computer so it's all new to me.

**Reply by osalbahr** [reply]
1. `(echo; echo 'eval "$(/usr/local/bin/brew shellenv)"') >> /Users/TJ/.profile`
2. `eval "$(/usr/local/bin/brew shellenv)"`

**Reply by JulianPatty** [reply]
Thank you very much, this helped a lot

**Comment by governor22** [comment]
I also get the same syntax error.  Do we need to run these "next steps" commands or will brew work OK without them ?

**Reply by osalbahr** [reply]
The first command is optional (and recommended) it is to add the `eval` command to your shell’s startup file.

The second command is needed at the start of each Terminal window. If you did add the `eval` command to the startup file, no need to manually type the command in any new Terminal window.

**Comment by amallio** [comment]
I get the same syntax error.

**Reply by osalbahr** [reply]
Could you copy-paste the full command and its error output?

**Comment by amallio** [comment]
Thanks osalbahr, I’m all set. I’m not exactly sure how my fix played out. It was late. Thanks for following up, though.

From: Osama Albahrani ***@***.***>
Date: Wednesday, January 17, 2024 at 2:54 AM
To: Homebrew/discussions ***@***.***>
Cc: Andy Mallio (amallio) ***@***.***>, Comment ***@***.***>
Subject: Re: [Homebrew/discussions] Telling me to run commands as "next steps" - where and how? (Discussion #4741)

Could you copy-paste the full command and its error output?

**Comment by Dhanraj-bidchat** [comment]
You have to add two commands:
first copy and paste this command on terminal and press Enter.
` (echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> /Users/yourHome/.zprofile`
 
 next do the same step for command
`  eval "$(/opt/homebrew/bin/brew shellenv)"`
  
  these two commands will successfully Homebrew to your PATH
  
  Next, check verify brew by following command
`  brew help`

**Comment by TucanoUrbano** [comment]
I am also having problems where the terminal seems to not accept what i am writing and always gives me the same error as i AM trying to complete these next steps. Here's the full list of code and errors, any help would be greatly appreciated
Next steps:
- Run these two commands in your terminal to add Homebrew to your PATH:
    (echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> /Users/pietrolavino/.zprofile
    eval "$(/opt/homebrew/bin/brew shellenv)"
- Run brew help to get started
- Further documentation:
    https://docs.brew.sh

+ brew install elan-init
/bin/bash: line 13: brew: command not found
pietrolavino@dhcp-128-189-94-212 ~ % 
pietrolavino@dhcp-128-189-94-212 ~ %  (echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> /Users/pietrolavino/.zprofile
    eval "$(/opt/homebrew/bin/brew shellenv)"
pietrolavino@dhcp-128-189-94-212 ~ % (echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv
subsh quote> echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv
subsh dquote cmdsubst> eval "$(/opt/homebrew/bin/brew shellenv)"
subsh dquote cmdsubst> echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> /Users/pietrolavino/.zprofile
    eval "$(/opt/homebrew/bin/brew shellenv)"
subsh dquote> (echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"'
subsh dquote> brew help
subsh dquote> "brew help"
subsh dquote> echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"
subsh dquote> echo;
subsh dquote> 
subsh dquote> echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"')
subsh dquote> (echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> /Users/pietrolavino/.zprofile

subsh dquote> nano /Users/pietrolavino/.zprofile
subsh dquote> 
subsh dquote> nano ~/.zprofile

subsh dquote> "nano ~/.zprofile
"
subsh dquote> 'nano ~/.zprofile
'
subsh dquote> nano
subsh dquote> brew help
subsh dquote> "brew help"
subsh dquote> echo;
subsh dquote> echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)" ')
subsh dquote> 
subsh dquote> miao
subsh dquote> "miao"
subsh dquote> 'eval "$(/opt/homebrew/bin/brew shellenv)"'
subsh dquote> /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
subsh dquote> pietrolavino@dhcp-128-189-94-212 ~ %  (echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> /Users/pietrolavino/.zprofile
subsh dquote> ~ %  (echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> /Users/pietrolavino/.zprofile 
subsh dquote> nano ~/.zprofile
subsh dquote> 'nano ~/.zprofile
subsh dquote>  'nano ~/.zprofile'
subsh dquote> touch ~/.zprofile
subsh dquote> 'touch ~/.zprofile'
subsh dquote> /Users/pietrolavino
subsh dquote> '/Users/pietrolavino'
subsh dquote> "/Users/pietrolavino"
subsh dquote> "$(/opt/homebrew/bin/brew shellenv)"') >> /Users/pietrolavino/.zprofile
    eval "$(/opt/homebrew/bin/brew shellenv)" 
subsh dquote>

**Reply by osalbahr** [reply]
Does your `~/.zprofile` contain `eval "$(/opt/homebrew/bin/brew shellenv)"`? If so, does the error persist after starting a new Terminal?

**Comment by w8sl** [comment]
Why just not add to the installation script (it is only required on arm64):
```
if [[ "$(uname -m)" == "arm64" ]]; then
   
   if ! [ -x "$(command -v brew)" ]; then
   
   echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> /Users/$USER/.zprofile
   eval "$(/opt/homebrew/bin/brew shellenv)"
   
   fi
fi
```

**Reply by SMillerDev** [reply]
Because it's important for us not to mess with the users files.
